src/main/java/com/aws/greengrass/deployment/DeviceConfiguration.java

```diff
@@ -328,7 +328,7 @@ void initializeNucleusLifecycleConfig(String nucleusComponentName, ComponentReci
     void initializeNucleusVersion(String nucleusComponentName, String nucleusComponentVersion) {
         kernel.getConfig().lookup(SERVICES_NAMESPACE_TOPIC, nucleusComponentName,
                 VERSION_CONFIG_KEY).dflt(nucleusComponentVersion);
-        kernel.getConfig().lookup(SETENV_CONFIG_NAMESPACE, GGC_VERSION_ENV).dflt(nucleusComponentVersion);
+        kernel.getConfig().lookup(SETENV_CONFIG_NAMESPACE, GGC_VERSION_ENV).withValue(nucleusComponentVersion);
```

Comment:
use `overrideValue` so that the timestamp is ignored and it always uses the value you give it.

Reply:
Why would we want to retain the timestamp though? that's what `overrideValue` does. `withValue` uses the current timestamp which I think is more accurate

Reply:
Because the timestamp isn't important. If the system clock is wrong, then this value may not be updated which is why I added the override method.